poseidon/src/lib.rs

```diff
@@ -0,0 +1,46 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+mod parameters;
+
+type CircuitField = ark_bls12_381::Fr;
+type FpVar = ark_r1cs_std::fields::fp::FpVar<CircuitField>;
+
+pub mod hash {
+    use ark_bls12_381::Fr;
+    use ark_std::vec;
+    use poseidon_permutation::Instance;
+
+    use crate::parameters::RATE_1_PARAMETERS;
+    /// hashes one field value, outputs a fixed length field value
+    pub fn one_to_one_hash(domain_separator: &Fr, value: Fr) -> Fr {
+        let parameters = RATE_1_PARAMETERS.clone();
+        let mut state = Instance::new(&parameters);
+        state.n_to_1_fixed_hash(vec![*domain_separator, value])
+    }
+}
+
+pub mod r1cs {
```

Comment:
Is this name standard for this level of abstraction? I'd go into either `circuit` or `constrained` directions but don't know standard nomenclature here yet.

Reply:
went with circuit, indeed perhaps we will go away from expressing them in r1cs